src/app/services/user-info-firebase.service.ts

```diff
@@ -42,11 +42,30 @@ export class UserInfoFirebaseService extends UserInfoService {
 
   override saveUser(user: UserData): Observable<string> {
     sessionStorage.setItem(user.uid,JSON.stringify(user));
-    return defer(() => from(setDoc(doc(collection(this.firestore, 'users'), user.uid), {
+    let userid = defer(() => from(setDoc(doc(collection(this.firestore, 'users'), user.uid), {
       uid: user.uid,
       ownedSets: user.getOwnedSetsToStore(),
       recentSets: user.getRecentSetsToStore()
     }) as Promise<void> ))
       .pipe(map((ret => user.uid as string)));
+
+      //Save user doesn't save to firebase unless this is here, it's 6:00am
+      userid.subscribe(a =>{
+        console.log("Save User")
+      })
```

Comment:
This occurs due to the lazy nature of observables. They won't complete unless there is a subscriber that consumes the observable's output. I suggest moving the subscribe out of saveUser and into updateViewDate like so
``` ts
  override updateViewDate(studySet: StudySetData){
    this.loadUser(sessionStorage.getItem("uid")!)
    .pipe(take(1)).subscribe(user =>{
      user.updateRecentSets({setId: studySet.id! , viewed: new Date()});
      if(studySet.owner){
        if (studySet.owner === sessionStorage.getItem("uid")){
        user.updateOwned({setId: studySet.id! , viewed: new Date()});
        }
      }
      this.saveUser(user).subscribe();
    })
  }
```